File: packages/backend/src/app/entities/index-op.ts
import { Column, Entity } from 'typeorm';
import { BaseEntity } from './base';

@Entity()
export class IndexOperation extends BaseEntity {
  @Column()
  type: 'website' | 'file';

  @Column()
  status: 'running' | 'queued' | 'completed' | 'failed' | 'cancelled';

  @Column()
  metadata: string;
}

File: mobile/lib/src/models/index_op.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'index_op.freezed.dart';
part 'index_op.g.dart';

enum IndexOperationType {
  WEBSITE,
  FILE,
  WEBPAGE,
}

enum IndexOperationStatus {
  FAILED,
  SUCCEEDED,
  RUNNING,
  COMPLETED,
}

@freezed
class IndexOperation with _$IndexOperation {
  const IndexOperation._();

  factory IndexOperation({
    required String id,
    required DateTime createdAt,
    required DateTime updatedAt,
    required IndexOperationType type,
    required IndexOperationStatus status,
    Map<String, dynamic>? metadata,
  }) = _IndexOperation;

  @override
  // ignore: recursive_getters
  DateTime get createdAt => createdAt.toLocal();

  @override
  // ignore: recursive_getters
  DateTime get updatedAt => updatedAt.toLocal();

  factory IndexOperation.fromJson(Map<String, dynamic> json) => _$IndexOperationFromJson(json);
}
